primitives/core/src/offchain/testing.rs

```diff
@@ -61,6 +62,58 @@ pub struct PendingRequest {
 	pub response_headers: Vec<(String, String)>,
 }
 
+/// Sharable "persistent" offchain storage for test.
+#[derive(Debug, Clone, Default)]
+pub struct TestPersistentOffchainDB {
+	persistent: Arc<RwLock<InMemOffchainStorage>>,
+}
+
+impl TestPersistentOffchainDB {
+	/// Create a new and empty offchain storage db for persistent items
+	pub fn new() -> Self {
+		Self {
+			persistent: Arc::new(RwLock::new(InMemOffchainStorage::default()))
+		}
+	}
+
+	/// Apply a set of off-chain changes directly to the test backend
+	pub fn apply_offchain_changes(&mut self, changes: &mut OffchainOverlayedChanges) {
+		let mut me = self.persistent.write();
+		for ((_prefix, key), value_operation) in changes.drain() {
```

Comment:
I'm a bit confused about the `prefix`:
1. Why is it dropped here? 
2. Why isn't prefix merged with `key` in some of the upper layers? `prefix` seems like something very high-level, and on the KVDB level we should really be left with only `keys` and `values`.
3. The prefix logic here introduces inconsistency with real-world backend and I think it would be best to avoid that.

Reply:
Issue here is that `Test*` assumes a prefix of `""` but the non-`Test*` structures use `"storage"` as prefix - to avoid this disconnect, the prefis is dropped here and a few lines down `""` is used instead.